## Problem 1a

import pandas as pd
import matplotlib.pyplot as plt

### Retrieve data from URL and create a dataframe

In [56]:
googleSheetId = '16i38oonuX1y1g7C_UAmiK9GkY7cS-64DfiDMNiR41LM'
worksheetName = 'Sheet1'
URL = 'https://docs.google.com/spreadsheets/d/%s/gviz/tq?tqx=out:csv&sheet=%s' % (googleSheetId,worksheetName)
df = pd.read_csv(URL)
df.head()

#add a field representing the price of each pair of shoes
df['price_per_pair'] = df['order_amount']/df['total_items']

### Get an overview of the dataset

In [57]:
print(df.describe())

df.info()


          order_id      shop_id      user_id   order_amount  total_items  \
count  5000.000000  5000.000000  5000.000000    5000.000000   5000.00000   
mean   2500.500000    50.078800   849.092400    3145.128000      8.78720   
std    1443.520003    29.006118    87.798982   41282.539349    116.32032   
min       1.000000     1.000000   607.000000      90.000000      1.00000   
25%    1250.750000    24.000000   775.000000     163.000000      1.00000   
50%    2500.500000    50.000000   849.000000     284.000000      2.00000   
75%    3750.250000    75.000000   925.000000     390.000000      3.00000   
max    5000.000000   100.000000   999.000000  704000.000000   2000.00000   

       price_per_pair  
count     5000.000000  
mean       387.742800  
std       2441.963725  
min         90.000000  
25%        133.000000  
50%        153.000000  
75%        169.000000  
max      25725.000000  
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5000 entries, 0 to 4999
Data columns (total 8 col

As can be seen from data description, the $3145.13 average order value (AOV) is the mean of the field 'order_amount'. 

The standard deviation for this field is $41282.539349. The standard deviation indicates how spread out the data are from the mean value. In this case, the standard deviation is very large, meaning the data points are spread out over a large range of values. While the 75th percentile is 390, the maximum value is 704000 in this field, meaning that there are large outliers.The shop with maximum 'order_amount' sells 2000 each order while the number of shoes in other orders are much smaller. Therefore, I assume the shop with maximum 'order_amount' are doing some types of wholesales while the rest are retailers. 

Similarly, the standard deviation for the 'price_per_pair' field is very large ($2441.963725), and the difference in the 75 percentile ($169) and the maximum value ($25725) is huge. Therefore, there are also outlier in the shoes price among the shops. I assume the shop that sells $25725 shoes sell hype shoes. This outliers also affect the 'order_amount' field.

It is not reasonable to calculate the mean because the data contains so many outliers. Nothing is wrong with the calculation, but it is not a good way to evaluate this dataset.

A better way to evaluate this data is to use median instead of mean to get the Average Order Value(AOV)


# Problem 1b

I would still use the 'order_amount' attribute to report for this dataset. As the goal is to do some analysis on the Average Order Value, this is the most appropriate attribute to use.

# Problem 1c

In [58]:
df['order_amount'].median()

284.0

The median value of the 'order_amount' metric is 284, so the value I need to find is $284